addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3045,12 +3057,24 @@ const SnippetOptionWidget = Widget.extend({
         // (otherwise we suppose that we should use the actual related color).
         if (params.colorNames && params.colorPrefix) {
             const classes = weUtils.computeColorClasses(params.colorNames, params.colorPrefix);
-            this.$target[0].classList.remove(...classes);
+            const isColorCombination = weUtils.isColorCombinationName(widgetValue);
+            if (params.colorThemeCombination) {
```

Comment:
I don't understand the meaning of that parameter from its name

Reply:
It should be something like `areColorAndThemeCombinable`.
It is only the case if this setting is enabled... but if I change the mechanism to separate `setStyle` from the other two, then this should disappear as it becomes the only behavior.